### ISSUE TITLE
Functions - Implementations Matrix

### ISSUE THREAD

**Issue by styk-tv** [RedisGraph/RedisGraph#433]
@jeffreylovitz 

It would be ideal (perhaps I'm not aware this already exists somewhere) to start a table that represents functions currently implemented and priority for implementation into next versions. 

Functions like `keys()` or `properties()` are of extreme importance but we should all have at least visibility as to what is planned when. I started a table under Wiki section.

https://github.com/RedisLabsModules/RedisGraph/wiki/Functions-Implementation-Matrix

**Comment by jeffreylovitz** [comment]
Thanks, @styk-tv! I agree that documentation on coverage is extremely important, and rather lacking right now (also, apparently, a bit too hard to find!).

The current home for this is (https://oss.redislabs.com/redisgraph/cypher_support/)[https://oss.redislabs.com/redisgraph/cypher_support/], but I think that your tabular format is more readable. Also, the reference material I used for writing this page seems to have more holes than I'd realized, especially in areas that have been volatile in Cypher (like predicate functions).

I think it makes more sense to continue using that site as our source of truth for documentation like this, but the format, visibility, and correctness could obviously use improvement. You can feel free to open PRs there (it's all populated by the `docs` folder in this repo, and can be tested with the `make localdocs` command), or I'll try to get to it soon!

**Comment by styk-tv** [comment]
Momentum picking up: #448 #447 #446 #449

Primary index matrix would be great but also consideration for APOC procedures/functions (if possible). A lot of the gap in primary index of required functions specified in wiki page could be satisfied when possibility of user-defined functions is allowed.

**Comment by jeffreylovitz** [comment]
User-defined functions is one of our major goals for the...medium term, I suppose? Tentatively, I'd say that we hope to deliver this in the next 3-4 months, but that number is _highly_ subject to change.

We started drafting an API for non-Cypher manipulations of graphs with #337. The implementation of this, actually, is what I had alluded to in #347!

Other priorities impeded this for the past few months, but we should be back on track within a few weeks.
